packages/server/modules/shared/test/authz.spec.ts

```diff
@@ -9,167 +9,227 @@ const {
   allowForRegisteredUsersOnPublicStreamsEvenWithoutRole,
   allowForServerAdmins,
   validateResourceAccess,
-  validateRequiredStreamFactory
-} = require('@/modules/shared/authz')
-const {
-  ForbiddenError: SFE,
-  UnauthorizedError: SUE,
+  validateRequiredStreamFactory,
+  AuthContext,
+  isAuthFailedResult,
+  AuthFailedResult
+} from '@/modules/shared/authz'
+import {
+  ForbiddenError as SFE,
+  UnauthorizedError as SUE,
   UnauthorizedError,
   ContextError,
-  NotFoundError
-} = require('@/modules/shared/errors')
-const { Roles } = require('@speckle/shared')
-const {
-  TokenResourceIdentifierType
-} = require('@/modules/core/graph/generated/graphql')
-const { ProjectRecordVisibility } = require('@/modules/core/helpers/types')
+  NotFoundError,
+  BaseError
+} from '@/modules/shared/errors'
+import { AvailableRoles, ensureError, Roles } from '@speckle/shared'
+import { TokenResourceIdentifierType } from '@/modules/core/graph/generated/graphql'
+import { ProjectRecordVisibility, StreamRecord } from '@/modules/core/helpers/types'
+import {
+  AuthData,
+  AuthPipelineFunction,
+  AuthResult
+} from '@/modules/shared/domain/authz/types'
+import { UserRoleData } from '@/modules/shared/domain/rolesAndScopes/types'
 
 describe('AuthZ @shared', () => {
+  const buildFooAuthData = (): AuthData =>
```

Comment:
This one was the hardest to migrate, there are a lot of places where the test feeds in incomplete date, hence a lot of forced type casts.

Hopefully we can get rid of the auth pipeline soon (use auth policies) and then get rid of this too

Reply:
agreed